### PR TITLE
Remove all Lua Policy Engine code

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
As per discussion with @habbie, this request drops all code and tests related to LPE.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: docs/WIP/luapolicy.xml [hunk 1]
```diff
@@ -1,9 +1,12 @@
-    <sect1 id="lua-policy-engine"><title>Lua Policy Engine</title>
+    <sect1 id="lua-policy-engine"><title>Lua Policy Engine (OBSOLETE)</title>
     <para>
       Starting with release 3.4.0, the PowerDNS Authoritative Server has support for Lua scripting to
       make policy decisions. The most common usage for these hooks is to implement RRL or RRL-like policies.
     </para>
     <para>
+      Since release 4.1.0, the PowerDNS Authoritative Server no longer supports this setting.
```
Comment (Habbie) [inline docs/WIP/luapolicy.xml]: mention dnsdist here

File: pdns/dnspacket.cc [hunk 2]
```diff
@@ -360,12 +360,6 @@ void DNSPacket::wrapup()
     addTSIG(pw, d_trc, d_tsigkeyname, d_tsigsecret, d_tsigprevious, d_tsigtimersonly);
   
   d_rawpacket.assign((char*)&packet[0], packet.size()); // XXX we could do this natively on a vector..
-
-  // copy RR counts so LPE can read them
```
Comment (Habbie) [inline pdns/dnspacket.cc]: I wonder if leaving this in would avoid surprises at a later time. It does not look like it hurts anyway.

**Comment by cmouse** [comment]
I forgot to put it here, but this requires https://github.com/ahupowerdns/luawrapper/pull/35 included in PowerDNS first. I should drop the commit from here.

File: docs/WIP/luapolicy.xml (hunk 1 above)
Comment (cmouse) [inline docs/WIP/luapolicy.xml]: dnsdist also had policy engine?

File: pdns/dnspacket.cc (hunk 2 above)
Comment (cmouse) [inline pdns/dnspacket.cc]: Restored the code segment.

**Comment by cmouse** [comment]
nits fixed.

**Comment by Habbie** [comment]
Sorry, when I asked to mention dnsdist in the docs, I did not realise this feature never even made it to the actual docs. Can you drop the xml completely, and put a note in the auth upgrade notes instead? Thanks!

**Comment by cmouse** [comment]
done

File: docs/markdown/authoritative/upgrading.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ Please upgrade to the PowerDNS Authoritative Server 4.0.0 from 3.4.2+. See the [
 
 ## Changed options
 
+ * `experimental-lua-policy-script` and feature has been dropped completely. We invite you to to use [(PowerDNS dnsdist)[http://dnsdist.org] instead.
```
Comment (Habbie) [inline docs/markdown/authoritative/upgrading.md]: s/and feature/option and feature/

s/to to use/to use/ and also i think the first [ needs to go

File: docs/markdown/authoritative/upgrading.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ Please upgrade to the PowerDNS Authoritative Server 4.0.0 from 3.4.2+. See the [
 
 ## Changed options
 
+ * `experimental-lua-policy-script` option and the feature itself has been dropped completely. We invite you to to use (PowerDNS dnsdist)[http://dnsdist.org] instead.
```
Comment (Habbie) [inline docs/markdown/authoritative/upgrading.md]: s/has been/have been/

s/to to use/to use/

**Pull request merged**